cmd/cli/certificate_rotate.go

```diff
@@ -0,0 +1,352 @@
+	}
+
+	if activeMrc == nil {
+		return nil, fmt.Errorf("not able to find MeshRootCertificate in active state")
+	}
+
+	return activeMrc, nil
+}
+
+func (r *rotateCmd) updateCertificate(name string, intent v1alpha2.MeshRootCertificateIntent) error {
+	mrc, err := r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	mrc.Spec.Intent = intent
+	_, err = r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Update(context.Background(), mrc, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *rotateCmd) createOrFindNewMRC(mrc *v1alpha2.MeshRootCertificate) (*v1alpha2.MeshRootCertificate, error) {
+	if r.mrcName != "" {
+		fmt.Fprintf(r.out, "using existing MeshRootCertificate %s\n", r.mrcName)
+
+		existing, err := r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Get(context.Background(), r.mrcName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("unable to verify that MeshRootCertificate [%s]. Error: %v", existing.Name, err)
+		}
+
+		if existing.Spec.Intent != v1alpha2.InactiveIntent {
+			return nil, fmt.Errorf("MRC provided must be in Inactive role")
+		}
+
+		return existing, nil
+	}
+
+	if r.mrcFilePath != "" {
+		fmt.Fprintf(r.out, "using file %s\n", r.mrcFilePath)
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		file, err := os.ReadFile(r.mrcFilePath)
+		if err != nil {
+			return nil, err
+		}
+		obj, _, err := decode([]byte(file), nil, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		newMRC, ok := obj.(*v1alpha2.MeshRootCertificate)
+		if !ok {
+			return nil, fmt.Errorf("file provided is not recognized as MeshRootCertificate")
+		}
+		if newMRC.Spec.Intent != v1alpha2.InactiveIntent {
+			return nil, fmt.Errorf("file provided must be in Inactive role")
+		}
+
+		existing, err := r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Get(context.Background(), newMRC.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("unable to verify that MeshRootCertificate [%s] doesn't already exist. Error: %v", newMRC.Name, err)
+		}
+		if existing != nil {
+			return nil, fmt.Errorf("cannot use MeshRootCertificate [%s] as it already exists in cluster", newMRC.Name)
+		}
+
+		return r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Create(context.Background(), newMRC, metav1.CreateOptions{})
+	}
+
+	return r.createFromExisting(mrc)
+}
+
+func (r *rotateCmd) createFromExisting(mrc *v1alpha2.MeshRootCertificate) (*v1alpha2.MeshRootCertificate, error) {
+	switch {
+	case mrc.Spec.Provider.Tresor != nil:
+		return r.createTressorFromExisting(mrc)
+	case mrc.Spec.Provider.Vault != nil:
+		// since we can't create the secrets in vault we require them to create a file
+		return nil, fmt.Errorf("please use -file option to provide a MeshRootCertificate Vault provider")
```

Comment:
Could the mrc name also be provided instead of the mrc file?

Reply:
yes!